core/src/smartcontracts/wasm.rs

```diff
@@ -629,111 +726,79 @@ impl<'wrld> Runtime<'wrld> {
             .map_err(|err| Error::Decode(err.into()))
     }
 
-    /// Executes the given wasm smartcontract
-    ///
-    /// # Errors
-    ///
-    /// - if unable to construct wasm module or instance of wasm module
-    /// - if unable to find expected main function export
-    /// - if the execution of the smartcontract fails
-    pub fn execute(
-        &mut self,
-        wsv: &mut WorldStateView,
-        authority: AccountId,
-        bytes: impl AsRef<[u8]>,
-    ) -> Result<()> {
-        let span = wasm_log_span!("Smart contract execution", %authority);
-        let state = State::new(wsv, authority, self.config, span);
-
-        self.execute_smart_contract_with_state(bytes, state)
-    }
-
-    fn execute_smart_contract_with_state(
-        &mut self,
-        bytes: impl AsRef<[u8]>,
-        state: State,
-    ) -> Result<()> {
-        let mut store = self.create_store(state);
-        let smart_contract = self.create_smart_contract(&mut store, bytes)?;
+    #[allow(clippy::needless_pass_by_value)]
+    #[codec::wrap]
+    fn execute_query_as_validator(
```

Comment:
```suggestion
    fn execute_query(
```

why not just this? it's implicit in the type

Reply:
I've tried this, but it doesn't work. The reason is that original `execute_query` is defined for generic `Runtime` so there will be a conflict.

In other words technically for `Runtime<state::Validator>` exist both `execute_query` and `execute_query_as_validator`.

Reply:
> In other words technically for Runtime<state::Validator> exist both execute_query and execute_query_as_validator.

please don't do that. Maybe you can define a trait with default implementation. If not, you can have 3 `execute_query` methods defined on `Runtime<state::Trigger>`, `Runtime<state::Validator>`, `Runtime<state::SmartContract>`. If you're worried about code duplication you can define a freestanding method that you delegate to from `Runtime<state::Trigger>`, `Runtime<state::SmartContract>`

Reply:
Done